specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-06-02-preview/fleets.json

```diff
@@ -0,0 +1,817 @@
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerService/fleets": {
+      "get": {
+        "tags": [
+          "Fleets"
+        ],
+        "operationId": "Fleets_List",
+        "summary": "Lists fleets in the specified subscription.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/FleetListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List fleet resources in a given subscription": {
+            "$ref": "./examples/Fleets_ListBySub.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/fleets/{fleetName}/members/{fleetMemberName}": {
+      "put": {
+        "tags": [
+          "FleetMembers"
+        ],
+        "operationId": "FleetMembers_CreateOrUpdate",
+        "summary": "Creates or updates a fleet member.",
+        "description": "A member contains a reference to an existing Kubernetes cluster. Creating a member makes the referenced cluster join the Fleet.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FleetNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FleetMemberNameParameter"
+          },
+          {
+            "$ref": "#/parameters/IfMatchHeader"
+          },
+          {
+            "$ref": "#/parameters/IfNoneMatchHeader"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/FleetMember"
+            },
+            "description": "The Fleet member to create or update."
+          }
+        ],
+        "responses": {
```

Comment:
>responses [](http://example.com/codeflow?start=9&length=9)

Need to add 200 for final response.

Reply:
create is always asynchronous. following th eguidance we use 201 + provisioning state pattern

Respond to the initial PUT request with a 201 Created or 200 OK (per normal guidance).

https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put

Reply:
I should have said 200 is needed for PUT response when the resource already exists.

Reply:
adding 200